### PR TITLE
[37775] Work package in inconsistent state when deleting value already set to work package

### PR DESCRIPTION
**Pull request by HDinger**
- [x] Handle WPs before the budget is deleted
  - [x] Either reassign
  - [x] Or delete the budget from the WP 
- [x] Add tests

### CODE REVIEW

**Comment by HDinger** [comment]
Ticket: 
https://community.openproject.org/projects/openproject/work_packages/37775/activity

File: modules/budgets/app/controllers/budgets_controller.rb [hunk 1]
```diff
@@ -136,11 +137,21 @@ def update
   end
 
   def destroy
+    budget = Budget.find(params[:id])
+    if budget.work_packages.any?
+      redirect_to destroy_info_budget_path(budget)
```
Comment (oliverguenther) [inline modules/budgets/app/controllers/budgets_controller.rb]: I'd suggest to do that in a `before_action` to make the signature of destroy smaller. Then you also don't need the `return` statement.

File: modules/budgets/app/controllers/budgets_controller.rb [hunk 2]
```diff
@@ -136,11 +137,21 @@ def update
...
+      return
+    end
+
     @budgets.each(&:destroy)
     flash[:notice] = t(:notice_successful_delete)
     redirect_to action: 'index', project_id: @project
   end
 
+  def destroy_info
+   @possible_other_budgets = get_all_project_budgets - [@budget]
```
Comment (oliverguenther) [inline modules/budgets/app/controllers/budgets_controller.rb]: Here I'd prefer you use a query to exclude the current one: `@project.budgets.where.not(id: @budget.id)`

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Thanks for the extra mile to create the spec helper pages. I only have a few remarks

**Pull request merged**